### PR TITLE
adopt wx_serverd, udp_daemon and client from our code for the Sagittarius arm.

### PR DESCRIPTION
**Pull request by le-horizon**

### CODE REVIEW

**Comment by le-horizon** [comment]
@breakds do you know how to fix the CI build failure?
I've checked quickly with google and chatgpt but no luck for getting dynamixel sdk into dependencies.

**Comment by breakds** [comment]
> @breakds do you know how to fix the CI build failure? I've checked quickly with google and chatgpt but no luck for getting dynamixel sdk into dependencies.

I think we can probably ignore the CI for now. `dynamixel sdk` in the original setting is fetched via git submodule. We do not need that since we already have nix-based solution.

File: src/udp_daemon.cpp [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+#include "spdlog/spdlog.h"
+#include "interbotix_xs_driver/xs_logging.hpp" // Logging macros and utils
+#include "interbotix_xs_driver/xs_common.hpp"  // Common variables and types
+#include "interbotix_xs_driver/xs_driver.hpp"  // The InterbotixDriverXS class
+
+using boost::asio::ip::udp;
+
+namespace horizon
+{
+  namespace widowx
```
Comment (breakds) [inline src/udp_daemon.cpp:21]: How about `namespace horizon::widowx` to save some indentation real estate?

File: src/udp_daemon.cpp [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+    io_service_ = std::make_unique<boost::asio::io_service>();
+    socket_ =
+        std::make_unique<udp::socket>(*io_service_, udp::endpoint(udp::v4(), port_));
+  } catch (std::exception& e) {
+    spdlog::critical("Fatal error while creating the socket: {}", e.what());
+    std::exit(EXIT_FAILURE);
+  }
+
+  char *username;
+  username = getlogin();
```
Comment (breakds) [inline src/udp_daemon.cpp:188]: The indentation here is a bit messy. Shall we add `.clang-format`?

Comment (le-horizon) [reply src/udp_daemon.cpp:188]: Done.

File: src/udp_daemon.cpp (hunk 1 above)
Comment (le-horizon) [reply src/udp_daemon.cpp:21]: Done

File: src/udp_daemon.cpp [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    spdlog::error("getlogin() error");
+    std::exit(EXIT_FAILURE);
+  }
+  std::string path = "/home/";
+  path += username; path += "/interbotix_xs_driver";
+  std::string filepath_motor_configs, filepath_mode_configs;
+  filepath_motor_configs = path + "/wx250s_motor_config.yaml";
+  filepath_mode_configs = path + "/mode_configs.yaml";
+  bool write_eeprom_on_startup = true;
+  std::string logging_level = "DEBUG";
```
[the remark this answers is not in the document]
Comment (le-horizon) [reply src/udp_daemon.cpp:201]: Done.

**Review by le-horizon — COMMENTED** [review]
Thanks for the comments, fixed all.  PTAL.

Le

**Review by breakds — APPROVED** [review]

**Comment by breakds** [comment]
Thanks for the effort, Le!

**Pull request merged**